g11n-ws/vip-manager-l10n-agent/src/main/java/com/vmware/l10agent/service/SingleComponentServiceImpl.java

```diff
@@ -213,47 +217,184 @@ private boolean synchkey2VipL10n(RecordModel record,  String key, String srcValu
+		if(configs.getVipBaseL10nUrl().equalsIgnoreCase(PropertyContantKeys.LOCAL)) {
+			return true;
+		}
+
+		StringBuffer urlStr = new StringBuffer(configs.getVipBaseL10nUrl());
+		urlStr.append(PropertyContantKeys.L10N_SOURCE_COLLECT_SET_URL
+				.replace("{" + APIParamName.PRODUCT_NAME + "}", record.getProduct())
+				.replace("{" + APIParamName.VERSION + "}", record.getVersion())
+				.replace("{" + APIParamName.COMPONENT + "}", record.getComponent())
+				.replace("{" + APIParamName.LOCALE + "}", record.getLocale()));
+
+		List<KeySourceModel> sourceModels = new ArrayList<>();
+		int count =0;
+		for (Entry<String, Object> entry : model.getMessages().entrySet() ) {
+			KeySourceModel keySourceModel = new KeySourceModel();
+			keySourceModel.setKey(entry.getKey());
+			keySourceModel.setSource((String) entry.getValue());
+			sourceModels.add(keySourceModel);
+			count++;
+			if (count > configs.getSyncBatchSize()) {
+				logger.info("sync to l0n batch size: {}", count);
+				boolean reqResult = postBatchData(sourceModels, urlStr.toString());
+				if (reqResult){
+					count = 0;
+					sourceModels = new ArrayList<>();
+				}else {
+					return  false;
+				}
+			}
+
+		}
+
+		if (count>0){
+			logger.info("sync to l0n batch size: {}", count);
+			return postBatchData(sourceModels, urlStr.toString());
+		}
+
+		return true;
+	}
+	/**
+         * sync the local source content to internal l10n or i18n
+         */
 	@Override
 	public boolean synchComponentFile2Internal(RecordModel record) {
 
       ComponentSourceModel model =  getSourceComponentFile(record);
-      if (model != null) {
-		  for (Entry<String, Object> entry : model.getMessages().entrySet()) {
-			  try {
-				  boolean result = synchkey2VipL10n(record, entry.getKey(), (String) entry.getValue(), null, null);
-				  if (!result) {
-					  result = synchkey2VipI18n(record, entry.getKey(), (String) entry.getValue(), null, null);
-					  if (!result) {
-						  try {
-							  if (record.getStatus() < 5) {
-								  record.setStatus(record.getStatus() + 1);
-								  TaskSysnQueues.SendComponentTasks.put(record);
-							  }
-						  } catch (InterruptedException e) {
-
-							  Thread.currentThread().interrupt();
-
-							  logger.info(e.getMessage(), e);
-						  }
-						  return result;
-					  }
-				  }
-			  } catch (UnsupportedEncodingException e) {
-
-				  logger.error(e.getMessage(), e);
-				  return false;
-			  }
-		  }
+	  if (this.configs.isSyncBatchEnable()){
+		  return syncBatchSource(record, model);
+	  }else {
+		  return syncSingleSource(record, model);
 	  }
+
+	}
+
+	private  boolean syncBatchSource(RecordModel record, ComponentSourceModel model){
+		if (model != null ){
+			boolean l10nResult = syncBatchKey2VipL10n(record, model);
+			boolean i18nResult = syncBatchKey2VipI18n(record, model);
+
+			if (l10nResult && i18nResult){
+				return true;
+			}else {
+				try {
+					if (record.getStatus() < 5) {
```

Comment:
What's the reason to judge and add status by 1?

Reply:
retry if the network not good make  connect remote service failure